cpu/rpx0xx/periph/i2c.c

```diff
@@ -14,116 +14,556 @@
+    }
+    
+    //write reg address first
+    _i2c_bus[dev].cmd = 0;
+    i2c_write_bytes(dev, addr, &reg, sizeof(reg), 0);
+
+    //repeated start to indicate shift to read
+    gpio_write(I2C_SDA_PIN, 1);
+    i2c_delay(I2C_DELAY);
+    gpio_write(I2C_SCL_PIN, 1);
+    i2c_delay(I2C_DELAY);
+    gpio_write(I2C_SCL_PIN, 0);
+    gpio_write(I2C_SDA_PIN, 0);
+    i2c_delay(I2C_DELAY);
+
+    //read reg data
+    _i2c_bus[dev].cmd = 1;
+    int status = i2c_read_bytes(dev, addr, data, sizeof(reg), flags);
+
+    return status;
 }
 
 int i2c_write_bytes(i2c_t dev, uint16_t addr, const void *data,
                     size_t len, uint8_t flags)
 {
-    if (IS_USED(MODULE_PIO_I2C) && dev >= I2C_NUMOF) {
-        pio_i2c_bus_t *i2c = pio_i2c_get(dev - I2C_NUMOF);
-        return i2c ? pio_i2c_write_bytes(i2c->pio, i2c->sm, addr, data, len, flags) : -EINVAL;
+
+    assert((dev < I2C_NUMOF) && (data != 0) && (len > 0) && (len < 256));
+
+    if (flags & I2C_NOSTART) { //or I2C_ADDR10 ?
+        return -EOPNOTSUPP;
+    }
+
+    //reset TX/RX FIFO Q
+    for(int i = 0; i < 256; i++){
+        txbuffer[i] = 0;
+    }
+
+    DEBUG("i2c write %d bytes", len);
+
+    //prepare i2c bus
+    long* baseaddr = (long *) I2C0_BASE;
+    _i2c_bus[dev].len = 8; //?
+    _i2c_bus[dev].cmd = 0; // send write cmd ?
+    *(baseaddr + IC_DATA_CMD) = _i2c_bus[dev].cmd;
+
+    //write IC_ENABLE[0] 0 to disable DW_apb_i2c until configured
+    long* memaddr = (long *) I2C0_BASE;
+    *(memaddr + IC_ENABLE) = *(memaddr + IC_ENABLE) & 0xFFFE;
+
+    //for 7-bit addressing
+    if(!(flags & I2C_ADDR10)){
+        /* write max speed mode supported to IC_CON (bits 2:1)*/
+        uint8_t mode = 0x2; //0x1 = standard, 0x2 = FM/FM+, 0x3 = HSM 
+        uint8_t addrmode = 0x0; //0 for 7-bit, 1 for 10-bit
+        uint8_t addrdata = 0x41 | (mode << 1) | (addrmode << 4);
+        *(memaddr + IC_CON) = *(memaddr + IC_CON) | (0x57 & (addrdata));
+
+        /* write target addr for dev */
+        //consider bits 8,9,11,13 for gen call / start cmd gen
+        *(memaddr + IC_TAR) = (*(memaddr + IC_TAR) & 0x3FF) | dev;
+        /* enable dw_a2b_i2c */
+        *(memaddr + IC_ENABLE) = *(memaddr + IC_ENABLE) | 0x1;
+    }
+    else{
+        /* write max speed mode supported to IC_CON (bits 2:1)*/
+        uint8_t mode = 0x2; //0x1 = standard, 0x2 = FM/FM+, 0x3 = HSM 
+        uint8_t addrmode = 0x1; //0 for 7-bit, 1 for 10-bit
+        uint8_t addrdata = 0x41 | (mode << 1) | (addrmode << 4);
+        *(memaddr + IC_CON) = *(memaddr + IC_CON) | (0x57 & (addrdata));
+
+        /* write target addr for dev */
+        //consider bits 8,9,11,13 for gen call / start cmd gen
+        *(memaddr + IC_TAR) = (*(memaddr + IC_TAR) & 0x3FF) | dev;
+        /* enable dw_a2b_i2c */
+        *(memaddr + IC_ENABLE) = *(memaddr + IC_ENABLE) | 0x1;
+    }
+
+
+    //transmit byte sequence
+    //clear txbuffer and set to data
+    for(int i = 0; i < 256; i++){
+        txbuffer[i] = 0;
+    }
```

Comment:
this might be a memset

but since there is a memcpy in the next line it might not be needed